avaframe/com1DFAPy/SPHfunctionsCython.pyx

```diff
@@ -0,0 +1,1264 @@
+    # yn = np.where(np.isnan(yn), 0, yn)
+    z = z / norme
+    # zn = np.where(np.isnan(zn), 0, zn)
+  return x, y, z
+
+
+def normalizepy(x, y, z): # <-- small wrapper to expose normalize() to Python
+    # cdef double xx = x
+    # cdef double yy = y
+    # cdef double zz = z
+    return normalize(x, y, z)
+
+
+@cython.cdivision(True)
+cdef (double, double, double) croosProd(double ux, double uy, double uz, double vx, double vy, double vz):
+  """ Compute cross product of vector u = (ux, uy, uz) and v = (vx, vy, vz).
+  """
+  cdef double wx = uy * vz - uz * vy
+  cdef double wy = uz * vx - ux * vz
+  cdef double wz = ux * vy - uy * vx
+  return wx, wy, wz
+
+
+def croosProdpy(x, y, z, u, v, w): # <-- small wrapper to expose croosProd() to Python
+    return croosProd(x, y, z, u, v, w)
+
+
+cdef double scalProd(double ux, double uy, double uz, double vx, double vy, double vz):
+  """ Compute scalar product of vector u = (ux, uy, uz) and v = (vx, vy, vz).
+  """
+  return ux*vx + uy*vy + uz*vz
+
+def scalProdpy(x, y, z, u, v, w): # <-- small wrapper to expose scalProd() to Python
+    return scalProd(x, y, z, u, v, w)
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+cdef double getScalar(double x, double y, double[:, :] V, double csz):
+  """ Interpolate vector field from grid to single point location
+
+  Originaly created to get the normal vector at location (x,y) given the
+  normal vector field on the grid. Grid has its origin in (0,0).
+  Can be used to interpolate any vector field.
+  Interpolation using a bilinear interpolation
+
+  Parameters
+  ----------
+      x: float
+          location in the x location of desiered interpolation
+      y: float
+          location in the y location of desiered interpolation
+      Nx: 2D numpy array
+          x component of the vector field at the grid nodes
+      Ny: 2D numpy array
+          y component of the vector field at the grid nodes
+      Nz: 2D numpy array
+          z component of the vector field at the grid nodes
+      csz: float
+          cellsize of the grid
+
+  Returns
+  -------
+      nx: float
+          x component of the interpolated vector field at position (x, y)
+      ny: float
+          y component of the interpolated vector field at position (x, y)
+      nz: float
+          z component of the interpolated vector field at position (x, y)
+  """
+  cdef int Lx0, Ly0, Lx1, Ly1
+  cdef double Lx, Ly
+  cdef double xllc = 0.
+  cdef double yllc = 0.
+
+  # find coordinates in normalized ref (origin (0,0) and cellsize 1)
+  Lx = (x - xllc) / csz
+  Ly = (y - yllc) / csz
+
+  # find coordinates of the 4 nearest cornes on the raster
+  Lx0 = <int>Lx
+  Ly0 = <int>Ly
+  Lx1 = Lx0 + 1
+  Ly1 = Ly0 + 1
+  # prepare for bilinear interpolation
+  dx = Lx - Lx0
+  dy = Ly - Ly0
+  # using bilinear interpolation on the cell
+
+  cdef double v = V[Ly0, Lx0]*(1-dx)*(1-dy) + V[Ly0, Lx1]*dx*(1-dy) + V[Ly1, Lx0]*(1-dx)*dy + V[Ly1, Lx1]*dx*dy
```

Comment:
make this multiline

Reply:
you mean make intermediate variables or just split the line?